Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -283,7 +283,7 @@ open class RouteController: NSObject {
         if (newLegIndex != progress.legIndex) {
             progress.legIndex = newLegIndex
         }
-        if (newStepIndex != progress.currentLegProgress.stepIndex) {
+        if (newStepIndex != progress.currentLegProgress.stepIndex) && (newStepIndex < progress.currentLeg.steps.endIndex - 1) {
```

Comment:
I wonder why it is `(newStepIndex < progress.currentLeg.steps.endIndex - 1)` instead of `(newStepIndex < progress.currentLeg.steps.endIndex)`. If we check `RouteLegProgress.stepIndex` it has the following assert: `assert(stepIndex >= 0 && stepIndex < leg.steps.endIndex)`. Is it intentional to ignore the last index?

Reply:
Thanks for the suggestion, I'll correct this one